src/lakefs_spec/spec.py

```diff
@@ -572,29 +572,35 @@ def ls(
 
         return info
 
-    def _open(
+    def open(
         self,
         path: str | os.PathLike[str],
-        mode: Literal["rb", "wb", "xb"] = "rb",
+        mode: Literal["r", "rb", "rt", "w", "wb", "wt", "x", "xb", "xt"] = "rb",
+        pre_sign: bool = False,
+        content_type: str | None = None,
+        metadata: dict[str, str] | None = None,
+        autocommit: bool = True,
         **kwargs: Any,
     ) -> LakeFSIOBase:
         """
-        Dispatch a lakeFS file (local buffer on disk) for the given remote path for up- or downloads depending on ``mode``.
-
-        Internal only, called by ``AbstractFileSystem.open()``.
+        Dispatch a lakeFS file-like object (local buffer on disk) for the given remote path for up- or downloads depending on ``mode``.
 
         Parameters
         ----------
         path: str | os.PathLike[str]
             The remote path for which to open a local ``LakeFSFile``. Must be a fully qualified lakeFS URI.
-        mode: Literal["rb", "wb", "xb"]
-            The file mode indicating its purpose. Use ``rb`` for downloads from lakeFS, ``wb/xb`` for uploads to lakeFS.
+        mode: Literal["r", "rb", "rt", "w", "wb", "wt", "x", "xb", "xt"]
+            The file mode indicating its purpose. Use ``r/rb`` for downloads from lakeFS, ``w/wb/x/xb`` for uploads to lakeFS.
```

Comment:
```suggestion
            The file mode indicating its purpose. Use ``r/rb`` for downloads from lakeFS, ``w/wb/x/xb`` for uploads to lakeFS. Note: Text modes are Note: Text modes are internally treated as binary modes.
```

Reply:
I'm not in favor of this anymore, as the text mode is coerced to the corresponding normal mode (`rt->r`, `wt->w`,...), which is not as surprising as binary coercion (in fact, that is exactly the same behavior as in builtin open()).

Should we just roll with that without a note?